spirit-score/router.js

```diff
@@ -0,0 +1,226 @@
+    ]
+  });
+  return res.json(gameWithUpdatedSpirit);
+}
+function allowedToEditSpiritScore(spiritScoreFor, user, game, userRoleId) {
+  if (userRoleId === spiritCaptain || userRoleId === teamCaptain) {
+    if (
+      (spiritScoreFor === "home" && user.teamId === game.awayTeamId) ||
+      (spiritScoreFor === "away" && user.teamId === game.homeTeamId)
+    ) {
+      return true;
+    }
+  } else if (userRoleId === clubBoard) {
+    if (
+      (spiritScoreFor === "home" &&
+        game.awayTeam.organisationId === user.organisationId) ||
+      (spiritScoreFor === "away" &&
+        game.homeTeam.organisationId === user.organisationId)
+    ) {
+      return true;
+    }
+  } else return false;
+}
+
+const router = new Router();
+
+router.post("/spirit-scores", auth, async (req, res, next) => {
+  try {
+    // console.log("req.body from spirit scores POST:", req.body);
+    const admins = [federation, superAdmin];
+    const teamUsers = [spiritCaptain, teamCaptain, clubBoard];
+    const userRoleId = req.user.organisation
+      ? req.user.organisation.roleId
+      : req.user.roleId;
+
+    const gameToScore = await Game.findByPk(req.body.gameId);
+
+    const { RKUScore, FNBScore, FMScore, PASCScore, COMMScore } = req.body;
+    const spiritTotal = RKUScore + FNBScore + FMScore + PASCScore + COMMScore;
+
+    if (admins.includes(userRoleId)) {
+      if (req.body.spiritScoreFor === "home") {
+        if (gameToScore.homeTeamReceivedSpiritScoreId) {
+          return return409(res);
+        } else {
+          return createSpiritScoreInDbAndSendResponse(
+            req,
+            res,
+            spiritTotal,
+            gameToScore.homeTeamId,
+            "home"
+          );
+        }
+      } else {
+        if (gameToScore.awayTeamReceivedSpiritScoreId) {
+          return return409(res);
+        } else {
+          return createSpiritScoreInDbAndSendResponse(
+            req,
+            res,
+            spiritTotal,
+            gameToScore.awayTeamId,
+            "away"
+          );
+        }
+      }
+    } else if (teamUsers.includes(userRoleId)) {
+      if (req.user.teamId === gameToScore.homeTeamId) {
+        if (gameToScore.awayTeamReceivedSpiritScoreId) {
+          return return409(res);
+        } else {
+          return createSpiritScoreInDbAndSendResponse(
+            req,
+            res,
+            spiritTotal,
+            gameToScore.awayTeamId
+          );
+        }
+      } else if (req.user.teamId === gameToScore.awayTeamId) {
+        if (gameToScore.homeTeamReceivedSpiritScoreId) {
+          return return409(res);
+        } else {
+          return createSpiritScoreInDbAndSendResponse(
+            req,
+            res,
+            spiritTotal,
+            gameToScore.homeTeamId
+          );
+        }
+      } else {
+        return return403(res);
+      }
+    } else {
+      return return403(res);
+    }
+  } catch (error) {
+    next(error);
+  }
+});
```

Comment:
This is just so you know. As logic in the backend grows, like yours, we normally create a new "layer" of files, the business layer, which handles the logic of db updates and queries. In this case for example these helper functions you have around here would live there and also a lot of the routes logic. Let's say from this route you would call a function called `addSpiritScoreToTeam` and be done with it

Reply:
This would be a new file in this same folder called idk `queries.js` or something like it

Reply:
The more you can modularise this logic to be non route specific is better as you'll get smaller functions for more specific actions to the db and then call that from higher functions